templates/CRM/Contribute/Page/UserDashboard.tpl

```diff
@@ -129,13 +129,18 @@
                     {foreach from=$recurRows item=row key=id}
                         <tr class="{cycle values="odd-row,even-row"}">
                             <td><label>{$recurRows.$id.amount|crmMoney}</label>
-                                every {$recurRows.$id.frequency_interval} {$recurRows.$id.frequency_unit}
-                                for {$recurRows.$id.installments} installments
+                                {ts 1=$recurRows.$id.frequency_interval 2=$recurRows.$id.frequency_unit} every %1 %2{/ts}
+                                {if !empty($recurRows.$id.installments)}
+                                    {ts 1=$recurRows.$id.installments} for %1 installments{/ts}
+                                {/if}
```

Comment:
Nice on the rebase.

Just noticing the {ts} - there's been a trend to replace these with something more translation-friendly even though it means more code. When broken up like this and without much context and where part of it is a translatable option value with plurals (e.g. `every %1 %2`) it's harder to translate. Here's an example: https://github.com/civicrm/civicrm-core/blob/d497396acb640b85152c69b1f2a325d2da810f1e/templates/CRM/Contribute/Form/Contribution/Confirm.tpl#L101-L132